pygments/cmdline.py

```diff
@@ -216,88 +135,68 @@ def _print_list(what):
             print("    %s" % docstring_headline(cls))
 
 
-def main_inner(popts, args, usage):
-    opts = {}
-    O_opts = []
-    P_opts = []
-    F_opts = []
-    for opt, arg in popts:
-        if opt == '-O':
-            O_opts.append(arg)
-        elif opt == '-P':
-            P_opts.append(arg)
-        elif opt == '-F':
-            F_opts.append(arg)
-        opts[opt] = arg
-
-    if opts.pop('-h', None) is not None:
-        print(usage)
+def main_inner(parser, argns):
+    if argns.help:
+        parser.print_help()
         return 0
 
-    if opts.pop('-V', None) is not None:
-        print('Pygments version %s, (c) 2006-2021 by Georg Brandl.' % __version__)
+    if argns.V:
+        print('Pygments version %s, (c) 2006-2021 by Georg Brandl, Matthäus '
+              'Chajdas and contributors.' % __version__)
         return 0
 
     # handle ``pygmentize -L``
-    L_opt = opts.pop('-L', None)
-    if L_opt is not None:
-        if opts:
-            print(usage, file=sys.stderr)
+    if argns.L is not None:
+        if any(v for (k, v) in vars(argns).items() if k != 'L'):
+            parser.print_help(sys.stderr)
             return 2
-
         # print version
         main(['', '-V'])
-        if not args:
-            args = ['lexer', 'formatter', 'filter', 'style']
-        for arg in args:
+        largs = argns.L
+        if not largs:
```

Comment:
I'd change this to:
```python
items = {'lexer', 'formatter', 'filter', 'style'}
if any(arg not in items for arg in largs):
    parser.print_help(sys.stderr)
# continue as before
```

Rationale: `_print_list` has no error checking, and if you don't set it correctly, it will silently terminate.

Reply:
:+1:

Reply:
Done